### ISSUE TITLE
LLDB debug ClickHouse startup is too slow

### ISSUE THREAD

**Issue by dong524dong** [ClickHouse/ClickHouse#36185]
Hello，
I compiled the master branch with Clang14, but when debugging with LLDB, the startup is too slow. And when I add some comments to the source code, the last `Linking CXX executable programs/clickhouse` is also very time consuming.
Have any optimization suggestions? 
Thanks!

**Operating system**
ubuntu-18.04

**Cmake version**
3.21.1 

**Ninja version**
1.8.2 

**Compiler name and version**
clang+llvm-14.0.0-x86_64-linux-gnu-ubuntu-18.04

**computer hardware configuration**
CPU processor: 48
MEM total: 100G+

**Comment by nikitamikhaylov** [comment]
@dong524dong Add `-DENABLE_THINLTO=0` to the cmake command to make linkage more fast.

**Comment by nikitamikhaylov** [comment]
How much does debugging with LLDB takes in comparison with GDB?

**Comment by dong524dong** [comment]
> @dong524dong Add `-DENABLE_THINLTO=0` to the cmake command to make linkage more fast.
@nikitamikhaylov thanks a lot for your answer！

It should be a little bit better, but it's not obvious.
Here are my CMAKE parameters: `-G Ninja "-DCMAKE_C_COMPILER=$(command -v clang)" "-DCMAKE_CXX_COMPILER=$(command -v clang++)" -DCMAKE_BUILD_TYPE=Debug -DENABLE_TESTS=0 -DENABLE_UTILS=0 -DENABLE_THINLTO= 0 -DENABLE_NURAFT=0 -DDISABLE_HERMETIC_BUILD=1
`

**Comment by dong524dong** [comment]
> How much does debugging with LLDB takes in comparison with GDB?

Switching to GDB debugging, its startup time is also unbearably slow, basically the same as LLDB. Is it because this is compiled with CLANG?

But I tried compiling with GCC (although it is not officially recommended). After successful compilation, I used GDB to debug. The startup speed is a little slow, but it is acceptable, but it is not friendly to track variable values.
